src/atmega2560p/hal/power.rs

```diff
@@ -0,0 +1,250 @@
+    /// chip ATMEGA2560P.
+    /// All the clock features are implemented in this function using match cases.
+    pub fn disable_clocks(&mut self, mode: Options) {
+        let mut prr;
+        match mode {
+            Options::TWI
+            | Options::TIMER2
+            | Options::TIMER0
+            | Options::TIMER1
+            | Options::SPI
+            | Options::USART0
+            | Options::ADC => {
+                prr = unsafe { core::ptr::read_volatile(&mut self.prr0) };
+            }
+
+            Options::TIMER5
+            | Options::TIMER4
+            | Options::TIMER3
+            | Options::USART3
+            | Options::USART2
+            | Options::USART1 => {
+                prr = unsafe { core::ptr::read_volatile(&mut self.prr1) };
+            }
+        }
+        match mode {
+            Options::TWI => {
+                prr = prr | 0x80;
+            }
+            Options::TIMER2 => {
+                prr = prr | 0x40;
+            }
+            Options::TIMER0 => {
+                prr = prr | 0x20;
+            }
+            Options::TIMER1 => {
+                prr = prr | 0x08;
+            }
+            Options::SPI => {
+                prr = prr | 0x04;
+            }
+            Options::USART0 => {
+                prr = prr | 0x02;
+            }
+            Options::ADC => {
+                prr = prr | 0x01;
+            }
+            Options::TIMER5 => {
+                prr = prr | 0x20;
+            }
+            Options::TIMER4 => {
+                prr = prr | 0x10;
+            }
+            Options::TIMER3 => {
+                prr = prr | 0x08;
+            }
+            Options::USART3 => {
+                prr = prr | 0x04;
+            }
+            Options::USART2 => {
+                prr = prr | 0x02;
+            }
+            Options::USART1 => {
+                prr = prr | 0x01;
+            }
+        }
+        match mode {
+            Options::TWI
+            | Options::TIMER2
+            | Options::TIMER0
+            | Options::TIMER1
+            | Options::SPI
+            | Options::USART0
+            | Options::ADC => unsafe {
+                core::ptr::write_volatile(&mut self.prr0, prr);
+            },
+
+            Options::TIMER5
+            | Options::TIMER4
+            | Options::TIMER3
+            | Options::USART3
+            | Options::USART2
+            | Options::USART1 => unsafe {
+                core::ptr::write_volatile(&mut self.prr1, prr);
+            },
+        }
+    }
+
+    /// This is the function for enabling the clock system of your choice.
+    /// Here we would create a new element of the structure power
+    /// and it would be used to control various clock gating features of the
+    /// chip ATMEGA2560P.
+    /// All the clock features are implemented in this function using match cases.
+    pub fn enable_clocks(&mut self, mode: Options) {
```

Comment:
Please let the user know, what exactly is options i.e. what he has to pass in as second argument.

Reply:
Adjusted, BTW it was mentioned as another comment in the starting of the code also. I have briefed it here again.